File: tt.js
const { Telegraf } = require('telegraf');
const request = require('request');
const fs = require('fs');

// Token bot Telegram dari BotFather
const TOKEN = '7006660414:AAFztKCoXAmk9IV3vA6DIE73cpIH3mswAUw';

// Inisialisasi bot
const bot = new Telegraf(TOKEN);

// Fungsi untuk mengunduh video dari URL menggunakan Cobalt API
function downloadVideo(url) {
    return new Promise((resolve, reject) => {
        const api_url = 'https://api.cobalt.tools/api/json';
        const payload = { url, vCodec: 'av1', vQuality: '1080' };
        const headers = { 'Accept': 'application/json' };

        request.post({ url: api_url, json: payload, headers }, (err, response, body) => {
            if (err) {
                reject(`Gagal mengunduh video. Error: ${err}`);
            } else if (response.statusCode !== 200) {
                reject(`Gagal mengunduh video. Status code: ${response.statusCode}`);
            } else {
                if (body.status === 'redirect') {
                    resolve({ status: 'redirect', url: body.url });
                } else {
                    const video_url = body.url;
                    const video_name = `video_${Math.random().toString(36).substring(7)}.mp4`;
                    const file = fs.createWriteStream(video_name);

                    request.get(video_url)
                        .on('error', (err) => {
                            reject(`Gagal mengunduh video. Error: ${err}`);
                        })
                        .pipe(file)
                        .on('finish', () => {
                            resolve({ status: 'download', video_name });
                        });
                }
            }
        });
    });
}

// Middleware untuk menanggapi pesan /start
bot.start((ctx) => ctx.reply('Halo! Ketik /download <URL TikTok etc> untuk mengunduh video.'));

// Middleware untuk menanggapi pesan /download <URL>
bot.command('download', async (ctx) => {
    const url = ctx.message.text.split(' ')[1];
    if (!url) {
        return ctx.reply('Gunakan perintah /download dengan URL TikTok etc.');
    }

    try {
        const result = await downloadVideo(url);
        
        if (result.status === 'redirect') {
            ctx.reply(`Video dapat diunduh dari URL berikut: ${result.url}`);
        } else {
            const videoName = result.video_name;
            // Kirim video yang telah diunduh ke pengguna
            ctx.replyWithVideo({ source: videoName });
        }
    } catch (error) {
        console.error('Error:', error);
        ctx.reply(error); // Balas pengguna jika terjadi kesalahan
    }
});

// Mulai bot
bot.launch().then(() => {
    console.log('Bot is running');
}).catch((err) => {
    console.error('Error starting bot', err);
});

File: tt.py
import requests
import telebot

# Inisialisasi bot Telegram
TOKEN = '7006660414:AAFztKCoXAmk9IV3vA6DIE73cpIH3mswAUw'
bot = telebot.TeleBot(TOKEN)

# Fungsi untuk mengunduh video dari URL menggunakan Cobalt API
def download_video(url):
    api_url = 'https://api.cobalt.tools/api/json'
    payload = {
        'url': url
    }
    headers = {
        'Accept': 'application/json'
    }
    
    try:
        response = requests.post(api_url, json=payload, headers=headers)
        if response.status_code == 200:
            json_response = response.json()
            video_url = json_response.get('url')
            
            # Mengunduh video dari URL yang didapat
            video_response = requests.get(video_url, stream=True)
            if video_response.status_code == 200:
                # Menyimpan konten video ke dalam file lokal
                with open('video.mp4', 'wb') as f:
                    for chunk in video_response.iter_content(chunk_size=1024):
                        if chunk:
                            f.write(chunk)
                return True, 'Video berhasil diunduh'
            else:
                return False, f'Gagal mengunduh video. Status code: {video_response.status_code}'
        else:
            return False, f'Error: {response.status_code} - {response.text}'
    except requests.exceptions.RequestException as e:
        return False, f'Error: {e}'

# Handler untuk perintah /start
@bot.message_handler(commands=['start'])
def send_welcome(message):
    bot.reply_to(message, 'Halo! Ketik /download <URL TikTok> untuk mengunduh video.')

# Handler untuk perintah /download
@bot.message_handler(commands=['download'])
def download(message):
    chat_id = message.chat.id
    if len(message.text.split()) > 1:
        url = message.text.split()[1]
        success, response = download_video(url)
        if success:
            bot.reply_to(message, response)
            # Kirim video yang telah diunduh ke pengguna
            video = open('video.mp4', 'rb')
            bot.send_video(chat_id, video)
        else:
            bot.reply_to(message, response)
    else:
        bot.reply_to(message, 'Gunakan perintah /download dengan URL TikTok.')

# Jalankan bot
bot.polling()
